lib/internal/modules/esm/translators.js

```diff
@@ -103,7 +103,9 @@ function errPath(url) {
 }
 
 async function importModuleDynamically(specifier, { url }, assertions) {
-  return asyncESM.esmLoader.import(specifier, url, assertions);
+  return asyncESM.esmLoader.import(specifier,
+                                   asyncESM.esmLoader.baseURL(url),
```

Comment:
And if `url` is a bare specifier? Does `baseURL` handle that case?

Reply:
it would be resolved relative to the baseURL as well. This only affects policies currently, but this matches how import maps should work (we can't make an integration test really right now).